pulp_rpm/app/tasks/synchronizing.py

```diff
@@ -100,15 +100,10 @@ def get_repomd_file(remote, url):
     downloader = remote.get_downloader(url=urlpath_sanitize(url, "repodata/repomd.xml"))
 
     try:
-        result = downloader.fetch()
-    except ClientResponseError as exc:
-        if 404 == exc.status:
-            return
-    except FileNotFoundError:
+        return downloader.fetch()
+    except (ClientResponseError, FileNotFoundError):
```

Comment:
I was debating whether we should raise non-404 errors or just return `None` here. I have a weak opinion that the reason why we couldn't download the file doesn't necessarily matter?  This function is mostly used to check various mirrors, and one could imagine that in that instance we truly don't care why it failed and we do want to check all the other mirrors.

https://developer.mozilla.org/en-US/docs/Web/HTTP/Status#client_error_responses

Reply:
I would generally agree, however one case which bothers me is the case when all mirrors fail because of bad remote configuration, or because of firewall or some other network problems.  If we just ignore it, user will have no idea why all mirrors are failing.
Maybe we should still skip all, non-404 ones as well, but at least log the error code and message. What do you think?

Reply:
So i hit this while trying to reproduce this bug:  https://pulp.plan.io/issues/9024  

The proxy was erroring with 'proxy auth required', which is a very helpful debugging message (ignoring that bug, maybe i forgot to put in my proxy auth credentials).  

I think having the error (or some error) bubbled up to the api , is extremely helpful here.  Its 'easier' for the non-mirror case, since there's only one error we care about reporting.  For the mirror case, i agree it is harder.  I'm not sure if returning the last error would be okay, or we need to somehow combine them together?  I'm a little iffy on that, but for the single url case, i think its very important to return the error.